node/_fs/_fs_opendir_test.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+import {
+  assert,
+  assertEquals,
+  assertFalse,
+  assertInstanceOf,
+  assertThrows,
+} from "../../testing/asserts.ts";
+import { opendir, opendirSync } from "./_fs_opendir.ts";
+import { Buffer } from "../buffer.ts";
+import { assertCallbackErrorUncaught } from "../_utils.ts";
+
+Deno.test("[node/fs] opendir()", async (t) => {
+  const path = await Deno.makeTempDir();
+  const file = await Deno.makeTempFile();
+
+  await t.step(
+    "fails if encoding is invalid",
+    () =>
+      opendir(
+        path,
+        { encoding: "invalid-encoding" },
+        (err) => assertInstanceOf(err, TypeError),
+      ),
+  );
+
+  await t.step(
+    "fails if bufferSize is invalid",
+    () =>
+      opendir(
+        path,
+        { bufferSize: -1 },
+        (err) => assertInstanceOf(err, RangeError),
+      ),
+  );
+
+  await t.step(
+    "fails if directory does not exist",
+    () =>
+      opendir(
+        "directory-that-does-not-exist",
+        (err) => assertInstanceOf(err, Error),
+      ),
+  );
+
+  await t.step(
+    "fails if not a directory",
+    () =>
+      opendir(
+        file,
+        (err) => assertInstanceOf(err, Error),
+      ),
+  );
+
+  await t.step(
+    "passes if path is a string",
+    () =>
+      opendir(
+        path,
+        (err, dir) => {
+          assertEquals(err, null);
+          assert(dir);
+        },
+      ),
+  );
+
+  await t.step(
+    "passes if path is a Buffer",
+    () =>
+      opendir(
+        Buffer.from(path),
+        (err, dir) => {
+          assertFalse(err);
+          assert(dir);
+        },
+      ),
+  );
+
+  await t.step(
+    "passes if path is a URL",
+    () =>
+      opendir(
+        new URL(`file://` + path),
+        (err, dir) => {
+          assertFalse(err);
+          assert(dir);
+        },
+      ),
+  );
+
+  await t.step("passes if callback isn't called twice", async () => {
+    const importUrl = new URL("./_fs_opendir.ts", import.meta.url);
+    await assertCallbackErrorUncaught({
+      prelude: `import { opendir } from "${importUrl}"`,
+      invocation: `opendir("${path}", `,
```

Comment:
I think we need to do something like `path.replaceAll("\\", "\\\\")` here to avoid the error below in windows:

```
    error: AssertionError: actual: "error: Uncaught SyntaxError: Octal escape sequences are not allowed in strict mode.
          opendir("C:\Users\RUNNER~1\AppData\Local\Temp\766483fe", (err) => {
                                                        ^
        at <anonymous> (file:///C:/actions-runner/work/deno_std/deno_std/$deno$eval.js:3:53)
    " expected to contain: "Error: success"
```

Reply:
There, that should do it. I used `JSON.stringify()`.

Reply:
> I used JSON.stringify().

That's much better 👍